src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1324,8 +1361,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     //   before sending it over the terminal's connection.
     void ControlCore::PasteText(const winrt::hstring& hstr)
     {
+        using namespace ::Microsoft::Console::Utils;
+
+        auto filtered = FilterStringForPaste(hstr, CarriageReturnNewline | ControlCodes);
+        if (BracketedPasteEnabled())
+        {
+            filtered.insert(0, L"\x1b[200~");
+            filtered.append(L"\x1b[201~");
+        }
+
+        // It's important to not hold the terminal lock while calling this function as sending the data may take a long time.
+        _sendInputToConnection(filtered);

```

Comment:
The above block is 1:1 what used to be the `Terminal::WritePastedText` but now it's outside the lock. It's inlined because `Terminal` otherwise asserts that it's locked when `BracketedPasteEnabled` gets called.